Flaky io.camunda.zeebe.it.cluster.network.DefaultAdvertisedAddressIT test

### <!-- Component -->

<!-- Zeebe- -->

### Description

io.camunda.zeebe.it.cluster.network.DefaultAdvertisedAddressIT is flaky. We first need to figure out why, then fix it; if we cannot reproduce it, or fix it, then and ONLY then can we evaluate if we should remove it. We would ONLY remove it if it brings little value, but this is to be evaluated case by case by the team (at least 4 eyes principle).

For some links, go to: https://dashboard.int.camunda.com/goto/dfkd6tq2mji0wc?orgId=1

Filter the test class name by: io.camunda.zeebe.it.cluster.network.DefaultAdvertisedAddressIT

You can use the test runs to figure out the issue.

### Impact

Blocks our current CI pipeline from being deterministic.

Sample failing run:

```
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 43.41 s <<< FAILURE! -- in io.camunda.zeebe.it.cluster.network.DefaultAdvertisedAddressIT
Error:  io.camunda.zeebe.it.cluster.network.DefaultAdvertisedAddressIT.shouldFormClusterWithDefaultAdvertisedHost -- Time elapsed: 43.34 s <<< ERROR!
io.camunda.client.api.command.ClientException: org.apache.hc.core5.http.ConnectionClosedException: Connection closed by peer
	at io.camunda.client.impl.http.ApiCallback.failed(ApiCallback.java:84)
	at org.apache.hc.core5.concurrent.BasicFuture.failed(BasicFuture.java:166)
	at org.apache.hc.core5.concurrent.ComplexFuture.failed(ComplexFuture.java:79)
	at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$2.failed(InternalAbstractHttpAsyncClient.java:367)
	at org.apache.hc.client5.http.impl.async.AsyncRedirectExec$1.failed(AsyncRedirectExec.java:261)
	at org.apache.hc.client5.http.impl.async.ContentCompressionAsyncExec$1.failed(ContentCompressionAsyncExec.java:183)
	at org.apache.hc.client5.http.impl.async.AsyncHttpRequestRetryExec$1.failed(AsyncHttpRequestRetryExec.java:203)
	at org.apache.hc.client5.http.impl.async.AsyncProtocolExec$1.failed(AsyncProtocolExec.java:297)
	at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.failed(HttpAsyncMainClientExec.java:135)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.failed(ClientHttp1StreamHandler.java:300)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.terminate(ClientHttp1StreamDuplexer.java:198)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.shutdownSession(AbstractHttp1StreamDuplexer.java:167)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onInput(AbstractHttp1StreamDuplexer.java:358)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.inputReady(AbstractHttp1IOEventHandler.java:64)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.inputReady(ClientHttp1IOEventHandler.java:41)
	at org.apache.hc.core5.reactor.InternalDataChannel.onIOEvent(InternalDataChannel.java:139)
	at org.apache.hc.core5.reactor.InternalChannel.handleIOEvent(InternalChannel.java:51)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.processEvents(SingleCoreIOReactor.java:193)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.doExecute(SingleCoreIOReactor.java:140)
	at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:92)
	at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.apache.hc.core5.http.ConnectionClosedException: Connection closed by peer
	... 10 more
```

No clear issue yet, but tentatively, I think it's not a bug, just that sometimes we're a little bit too slow to get an answer, and a plain `join` in an `Awaitility` block just, well, blocks - it won't allow for retries or anything.

I'll need to check the actual application logs; better if I can reproduce it, but so far no luck.

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/cluster/network/DefaultAdvertisedAddressIT.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/cluster/network/DefaultAdvertisedAddressIT.java
@@ -12,14 +12,18 @@ import static io.camunda.application.commons.security.CamundaSecurityConfigurati
 import static io.camunda.configuration.beans.LegacySearchEngineSchemaManagerProperties.CREATE_SCHEMA_ENV_VAR;
 
 import io.camunda.client.CamundaClient;
+import io.camunda.configuration.SecondaryStorage.SecondaryStorageType;
 import io.camunda.container.cluster.CamundaCluster;
-import io.camunda.container.cluster.CamundaPort;
 import io.camunda.zeebe.qa.util.testcontainers.ZeebeTestContainerDefaults;
 import io.camunda.zeebe.test.util.asserts.TopologyAssert;
-import java.net.URI;
+import io.camunda.zeebe.test.util.testcontainers.ContainerLogsDumper;
 import java.time.Duration;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.assertj.core.api.Assertions;
 import org.awaitility.Awaitility;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
 
@@ -37,57 +41,58 @@ final class DefaultAdvertisedAddressIT {
           .withGatewaysCount(1)
           .withBrokersCount(1)
           .withEmbeddedGateway(false)
-          .withNodeConfig(node -> node.withAdditionalExposedPort(8080))
-          // explicitly unset the (advertised) host to force computing the default
-          .withBrokerConfig(
+          .withNodeConfig(
               node -> {
-                node.getEnvMap().remove("ZEEBE_BROKER_NETWORK_ADVERTISEDHOST");
-                node.getEnvMap().remove("ZEEBE_BROKER_NETWORK_HOST");
-                node.addEnv(CREATE_SCHEMA_ENV_VAR, "false");
-                node.addEnv(UNPROTECTED_API_ENV_VAR, "true");
-                node.addEnv(AUTHORIZATION_CHECKS_ENV_VAR, "false");
-              })
-          .withGatewayConfig(
-              node -> {
-                node.getEnvMap().remove("ZEEBE_GATEWAY_CLUSTER_ADVERTISEDHOST");
-                node.getEnvMap().remove("ZEEBE_GATEWAY_CLUSTER_HOST");
-                node.addEnv(CREATE_SCHEMA_ENV_VAR, "false");
-                node.addEnv(UNPROTECTED_API_ENV_VAR, "true");
-                node.addEnv(AUTHORIZATION_CHECKS_ENV_VAR, "false");
+                node.withUnifiedConfig(
+                    c -> {
+                      // explicitly unset the (advertised) host to force computing the default
+                      c.getCluster().getNetwork().setAdvertisedHost(null);
+                      c.getCluster().getNetwork().setHost(null);
+                      c.getData().getSecondaryStorage().setType(SecondaryStorageType.none);
+
+                      node.addEnv(CREATE_SCHEMA_ENV_VAR, "false");
+                      node.addEnv(UNPROTECTED_API_ENV_VAR, "true");
+                      node.addEnv(AUTHORIZATION_CHECKS_ENV_VAR, "false");
+                    });
               })
           .build();
+
+  @SuppressWarnings("unused")
+  @RegisterExtension
+  private final ContainerLogsDumper logsDumper = new ContainerLogsDumper(cluster::getNodes);
 
   @Test
   void shouldFormClusterWithDefaultAdvertisedHost() {
     // given
     final var clientBuilder =
         CamundaClient.newClientBuilder()
-            .restAddress(
-                URI.create(
-                    "http://localhost:"
-                        + cluster
-                            .getAvailableGateway()
-                            .getMappedPort(CamundaPort.GATEWAY_REST.getPort())));
+            .defaultRequestTimeout(Duration.ofSeconds(5))
+            .restAddress(cluster.getAvailableGateway().getRestAddress())
+            .grpcAddress(cluster.getAvailableGateway().getGrpcAddress())
+            .preferRestOverGrpc(true);
 
     try (final var client = clientBuilder.build()) {
       // when - then
       Awaitility.await("until topology is complete")
-          .atMost(Duration.ofSeconds(30))
+          .atMost(Duration.ofMinutes(2))
           .untilAsserted(
               () ->
-                  TopologyAssert.assertThat(client.newTopologyRequest().send().join())
+                  TopologyAssert.assertThat(
+                          client.newTopologyRequest().send().join(5, TimeUnit.SECONDS))
                       .isComplete(1, 1, 1));
 
       Awaitility.await("until a client request is successful")
-          .atMost(Duration.ofSeconds(30))
+          .atMost(Duration.ofMinutes(1))
           .untilAsserted(
               () ->
-                  client
-                      .newPublishMessageCommand()
-                      .messageName("test-message")
-                      .correlationKey("test-key")
-                      .send()
-                      .join());
+                  Assertions.assertThat(
+                          (Future<?>)
+                              client
+                                  .newPublishMessageCommand()
+                                  .messageName("test-message")
+                                  .correlationKey("test-key")
+                                  .send())
+                      .succeedsWithin(Duration.ofSeconds(5)));
     }
   }
 }

